src/frontends/pytorch/src/op/repeat_interleave.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+namespace {
+OutputVector generate_indices_from_repeats_tensor(std::vector<int64_t> repeats, NodeContext& context) {
+    OutputVector all_indices;
+    for (int i = 0; i < repeats.size(); i++) {
+        Shape indices_shape{static_cast<size_t>(repeats.at(i))};
+        std::vector<int64_t> indices_vec(repeats.at(i), i);
+        auto indices = context.mark_node(opset10::Constant::create(element::i64, indices_shape, indices_vec));
+        all_indices.push_back(indices);
+    }
+    return all_indices;
+};
+}  // namespace
+
+OutputVector translate_repeat_interleave(NodeContext& context) {
+    // constants
+    auto const_0 = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {0}));
+    auto const_1 = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {1}));
+    auto const_1_list = context.mark_node(opset10::Constant::create(element::i64, Shape{1}, {1}));
+    auto const_neg_1 = context.mark_node(opset10::Constant::create(element::i64, Shape{1}, {-1}));
+
+    // inputs
+    auto input = context.get_input(0);
+    std::shared_ptr<ov::Node> result;
+
+    try {
```

Comment:
Try catch is not required in this case, please avoid it. You need to get `reapeats` as non constant and after that try to do `dynamic_pointer_cast` to `Constant` if it casted fine you can get const value from it.

Reply:
done